### ISSUE TITLE
Make it compatible with TS 5

### ISSUE THREAD

**Issue by AlenJ-dmt** [roerohan/react-vnc#53]
### Is there an existing issue for this?

- [X] I have searched existing issues, it hasn't been reported yet

### Use case description

Newer projects are mostly created with TS 5 it will be nice to have support for it, without forcing peer installations

**Comment by sancelot** [comment]
```
npm install react-vnc
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: react-scripts@5.0.1
npm ERR! Found: typescript@5.4.5
npm ERR! node_modules/typescript
npm ERR!   peer typescript@">=4.2.0" from ts-api-utils@1.3.0
npm ERR!   node_modules/ts-api-utils
npm ERR!     ts-api-utils@"^1.3.0" from @typescript-eslint/eslint-plugin@7.11.0
npm ERR!     node_modules/@typescript-eslint/eslint-plugin
npm ERR!       dev @typescript-eslint/eslint-plugin@"^7.2.0" from the root project
npm ERR!     ts-api-utils@"^1.3.0" from @typescript-eslint/type-utils@7.11.0
npm ERR!     node_modules/@typescript-eslint/type-utils
npm ERR!       @typescript-eslint/type-utils@"7.11.0" from @typescript-eslint/eslint-plugin@7.11.0
npm ERR!       node_modules/@typescript-eslint/eslint-plugin
npm ERR!         dev @typescript-eslint/eslint-plugin@"^7.2.0" from the root project
npm ERR!   dev typescript@"^5.2.2" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peerOptional typescript@"^3.2.1 || ^4" from react-scripts@5.0.1
npm ERR! node_modules/react-scripts
npm ERR!   peer react-scripts@">=4.0.0" from react-vnc@1.0.0
npm ERR!   node_modules/react-vnc
npm ERR!     react-vnc@"*" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: typescript@4.9.5
npm ERR! node_modules/typescript
npm ERR!   peerOptional typescript@"^3.2.1 || ^4" from react-scripts@5.0.1
npm ERR!   node_modules/react-scripts
npm ERR!     peer react-scripts@">=4.0.0" from react-vnc@1.0.0
npm ERR!     node_modules/react-vnc
npm ERR!       react-vnc@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR!
npm ERR! For a full report see:
npm ERR! C:\Users\sancelot\AppData\Local\npm-cache\_logs\2024-05-28T09_46_51_449Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in: C:\Users\sancelot\AppData\Local\npm-cache\_logs\2024-05-28T09_46_51_449Z-debug-0.log
```

**Comment by roerohan** [comment]
Hey, I just saw this. I'll work on adding TS 5 compatibility as soon as I get some time. Thanks for reporting this.

**Comment by roerohan** [comment]
Hey @sancelot @AlenJ-dmt
I just released a new version of react-vnc after updating all the dependencies. Please take a look and let me know if it's working fine for you! Haven't gotten much time to test it out, and there were some breaking changes so I released it as v2.0.0
